### PR TITLE
I bip

### PR DESCRIPTION
**Pull request by tsorya**
Adding ibip functionality by adding relevant files to masters ignition.
After adding files to ignition, bootkube.sh will run coreos-install with this ignition

### CODE REVIEW

File: cmd/cluster-bootstrap/start.go [hunk 1]
```diff
@@ -44,6 +45,7 @@ func init() {
 	cmdStart.Flags().StringSliceVar(&startOpts.requiredPodClauses, "required-pods", defaultRequiredPods, "List of pods name prefixes with their namespace (written as <namespace>/<pod-prefix>) that are required to be running and ready before the start command does the pivot, or alternatively a list of or'ed pod prefixes with a description (written as <desc>:<namespace>/<pod-prefix>|<namespace>/<pod-prefix>|...).")
 	cmdStart.Flags().StringVar(&startOpts.waitForTearDownEvent, "tear-down-event", "", "if this optional event name of the form <ns>/<event-name> is given, the event is waited for before tearing down the bootstrap control plane")
 	cmdStart.Flags().BoolVar(&startOpts.earlyTearDown, "tear-down-early", true, "tear down immediate after the non-bootstrap control plane is up and bootstrap-success event is created.")
+	cmdStart.Flags().StringVar(&startOpts.clusterProfile, "cluster-profile", "", "The cluster profile.")
```
Comment (sttts) [inline cmd/cluster-bootstrap/start.go:48]: what is this? What happens with the file? Add proper docs.

File: pkg/start/start.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ const (
 	bootstrapPodsRunningTimeout = 20 * time.Minute
 	// how long we wait until the assets must all be created
 	assetsCreatedTimeout = 60 * time.Minute
+	SingleNodeProductionEdge = "single-node-production-edge"
```
Comment (sttts) [inline pkg/start/start.go:31]: What is this? It is no edge. Is it a profile? Please name it that way.

File: pkg/start/start.go [hunk 3]
```diff
@@ -127,7 +131,9 @@ func (b *startCommand) Run() error {
 	if err = waitUntilPodsRunning(ctx, client, b.requiredPodPrefixes); err != nil {
 		return err
 	}
-	cancel()
+	if b.clusterProfile != SingleNodeProductionEdge {
+		cancel()
+	}
```
Comment (sttts) [inline pkg/start/start.go:146]: Adding this logic here and the knowledge that it depends on the profile IMO is wrong. Instead make the behaviour the feature, i.e. the flag: `--terminte-early-when-pods-running`.

**Comment by sttts** [comment]
/hold

cluster-bootstrap should not know anything about profiles. Turn the profile logic into an explicit feature of cluster-bootstrap, i.e. something like `--terminate-early-after-pods-running=true`.

File: cmd/cluster-bootstrap/ibip.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/spf13/cobra"
+
+)
+
+var (
+	cmdIBip = &cobra.Command{
+		Use:          "ibip",
+		Short:        "Update the master ignition with control plane static pods files",
+		Long:         "",
+		PreRunE:      validateIBipOpts,
```
Comment (romfreiman) [inline cmd/cluster-bootstrap/ibip.go:15]: no ibip please :)

**Comment by eranco74** [comment]
closing this PR. there is a new updated PR targeting master
#46

**Pull request closed without merging**